stdio/file.c

```diff
@@ -141,6 +148,11 @@ static void file_free(FILE *file)
 	}
 
 	resourceDestroy(file->lock);
+
+	mutexLock(file_common.lock);
```

Comment:
first remove from the list, then start destroying. Right now there might be a race with `fflush(NULL)` which will try to acquire destroyed lock?